Update manylinux version

bmerry opened the pull request:
The old manylinux2014 no longer builds because CentOS 7 is EOL. Update to manylinux_2_28.

Also update versions in the Github Actions workflow - the provided cibuildwheel was too old to work, and there were warnings about other deprecated versions of actions.

bmerry committed:
Update manylinux version

The old manylinux2014 no longer builds because CentOS 7 is EOL. Update
to manylinux_2_28.

Also update versions in the Github Actions workflow - the provided
cibuildwheel was too old to work, and there were warnings about other
deprecated versions of actions.

amishatishpatel commented on pyproject.toml:
```diff
@@ -10,8 +10,8 @@ requires = [
 before-all = "manylinux/before_all.sh"
-manylinux-x86_64-image = "manylinux2014"
-manylinux-i686-image = "manylinux2014"
+manylinux-x86_64-image = "manylinux_2_28"
+manylinux-i686-image = "manylinux_2_28"
```
Apologies for potentially reading too much into this version update. I'm not familiar with manylinux so I figured I'd at least verify the version syntax. In reading the `README` discussion on versions at https://github.com/pypa/manylinux
> The manylinux project supports:
* `manylinux2014` images for `x86_64`, `i686`, `aarch64`, `ppc641e` and `s390x`.
* `manylinux_2_28` images for `x86_64`, `aarch64`, `ppc64le` and `s390x`

Should I be worried that I don't see `i686` architecture for `manylinux_2_28`? At least listed in that bullet point.

bmerry replied:
Good catch. I've never actually tried building i686 wheels (who still uses i686???). I should probably just delete this line.

amishatishpatel commented on manylinux/before_all.sh:
```diff
@@ -1,5 +1,5 @@
 set -e
 
-yum-config-manager --add-repo https://developer.download.nvidia.com/compute/cuda/repos/rhel7/x86_64/cuda-rhel7.repo
+yum config-manager --add-repo https://developer.download.nvidia.com/compute/cuda/repos/rhel8/x86_64/cuda-rhel8.repo
```
Did the syntax change in rhel8?
* I was confused when looking up https://man7.org/linux/man-pages/man1/yum-config-manager.1.html (and seeing it was still `yum-config-manager`)
* Ended up doing a `docker run redhat/ubi8` to check the syntax and it is indeed `yum config-manager ...`

bmerry replied:
It seemed to have changed - before I made this change I got a command not found error with manylinux_2_28.

amishatishpatel reviewed (commented):
One comment for clarification before I explicitly Approve.

bmerry committed:
Remove config for i686 wheel

manylinux_2_28 doesn't actually support i686, and I haven't been
building wheels for it anyway.
